### DISCUSSION TITLE
Github action security use

### DISCUSSION THREAD

**Discussion by DhirajBembade** [community/community#120828] [Code Security] [answered]
### Select Topic Area

Question

### Body

I need guidance for using SonarQube for CI/CD in GitLab. Any material links or suggestions would be really helpful.

**Comment by davevad93** [comment, marked as answer]
Hi @DhirajBembade , take a look at these links:
- [Using GitLab CI/CD with a GitHub repository](https://docs.gitlab.com/ee/ci/ci_cd_for_external_repos/github_integration.html)
- [GitLab integration](https://docs.sonarsource.com/sonarqube/latest/devops-platform-integration/gitlab-integration/)
- [Introduction to GitHub integration](https://docs.sonarsource.com/sonarqube/latest/devops-platform-integration/github-integration/introduction/)
- [Setting up the GitHub integration with SonarQube](https://docs.sonarsource.com/sonarqube/latest/devops-platform-integration/github-integration/setting-up-integration/)
- [Adding the SonarQube analysis to your GitHub Actions workflow](https://docs.sonarsource.com/sonarqube/latest/devops-platform-integration/github-integration/adding-sonarqube-analysis-to-your-workflow/)
- [Setting up your GitHub projects in SonarQube](https://docs.sonarsource.com/sonarqube/latest/devops-platform-integration/github-integration/setting-up-your-projects/)

**Reply by DhirajBembade** [reply]
thank you for the information it is helpful .

**Reply by ghost** [reply]
> Hi @DhirajBembade , take a look at these links:
> 
> * [Using GitLab CI/CD with a GitHub repository](https://docs.gitlab.com/ee/ci/ci_cd_for_external_repos/github_integration.html)
> * [GitLab integration](https://docs.sonarsource.com/sonarqube/latest/devops-platform-integration/gitlab-integration/)
> * [Introduction to GitHub integration](https://docs.sonarsource.com/sonarqube/latest/devops-platform-integration/github-integration/introduction/)
> * [Setting up the GitHub integration with SonarQube](https://docs.sonarsource.com/sonarqube/latest/devops-platform-integration/github-integration/setting-up-integration/)
> * [Adding the SonarQube analysis to your GitHub Actions workflow](https://docs.sonarsource.com/sonarqube/latest/devops-platform-integration/github-integration/adding-sonarqube-analysis-to-your-workflow/)
> * [Setting up your GitHub projects in SonarQube](https://docs.sonarsource.com/sonarqube/latest/devops-platform-integration/github-integration/setting-up-your-projects/)

**Comment by Rifahad** [comment]
s:

[Using GitLab CI/CD with a GitHub repository](https://docs.gitlab.com/ee/ci/ci_cd_for_external_repos/github_integration.html)

**Reply by Rameeskc** [reply]
i got it

**Comment by gitricko** [comment]
You might wanna try sonarless which automates the above in just a few steps.

Local install:

Install CLI: `curl -s "https://raw.githubusercontent.com/gitricko/sonarless/main/install.sh" | bash`

Scan current directory: `sonarless scan`

Get results: `sonarless results`

Check out the automation

https://github.com/gitricko/sonarless

It even have a GitHub Action for GITHUB CI. The nice thing about this is that you don't need a host sonarqube
